connectors/sources/salesforce.py

```diff
@@ -33,6 +33,8 @@
 QUERY_ENDPOINT = f"/services/data/{API_VERSION}/query"
 DESCRIBE_ENDPOINT = f"/services/data/{API_VERSION}/sobjects"
 DESCRIBE_SOBJECT_ENDPOINT = f"/services/data/{API_VERSION}/sobjects/<sobject>/describe"
+# https://developer.salesforce.com/docs/atlas.en-us.244.0.api_rest.meta/api_rest/resources_sobject_blob_retrieve.htm
```

Comment:
```suggestion
# https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_blob_retrieve.htm
```

I guess we want to always point to the latest documentation

Reply:
I did this on purpose, because we hardcoded it to `v58.0`:

https://github.com/elastic/connectors/blob/56ad4513b3591f725452749c52e0ed5ca2c2e48c/connectors/sources/salesforce.py#L31

Reply:
v58 is also arbitrary, maybe we can bump it too?

It's IMO much worse to use 2 apis with different versions vs using newer API

Reply:
I can bump the version.

Actually, there is a [List Available REST API versions](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_versions.htm) API, but I think it may be overkill, and I also feel scary if we allow it to bump by itself.

Reply:
I bumped the version in https://github.com/elastic/connectors/pull/1800/commits/f018d551672c15dda81fe3182ae61a374816fe40, and then reverted it in https://github.com/elastic/connectors/pull/1800/commits/7596d654acc7bf766a6a695c1cba93fe08b0fd74.

Because there are many hardcoded `v58.0` in the test file, I think it's worth a separate PR. Drafting...

Reply:
A separate PR: https://github.com/elastic/connectors/pull/1807